virtool/indexes/api.py

```diff
@@ -236,3 +239,44 @@ async def find_history(req):
     )
 
     return json_response(data)
+
+
+async def reset_history(history: Collection, index_id: str):
+    """
+    Set the index.id and index.version fields with the given index id to 'unbuilt'.
+
+    :param history: The history collection of the database.
+    :param index_id: The ID of the index which failed to build
+    """
+    query = {
+        "_id": {
+            "$in": await history.distinct("_id", {"index.id": index_id})
+        }
+    }
+
+    return await history.update_many(query, {
+        "$set": {
+            "index": {
+                "id": "unbuilt",
+                "version": "unbuilt"
+            }
+        }
+    })
```

Comment:
Move to `virtool.indexes.db`. API modules should only contain handlers, route tables, and possibly constants.

Please pass the function `db` instead of the specific collection object.

1. This is convention in Virtool.
2. If in the future this function needs access to other collections, the function signature will not change and usages will not need to be updated.
3. `db.collection.method` is a pattern reflected in the `motor` documentation and Mongo shell